src/braket/default_simulator/openqasm/circuit.py

```diff
@@ -63,28 +64,47 @@ def num_qubits(self) -> int:
 
     @property
     def basis_rotation_instructions(self):
-        """
-        This function assumes all observables are commuting.
-        """
-
         basis_rotation_instructions = []
-        measured_qubits = set()
+        observable_map = {}
+
+        def process_observable(observable):
+            if isinstance(observable, Identity):
+                return
+            measured_qubits = tuple(observable.measured_qubits)
+            for qubit in measured_qubits:
+                for target, previously_measured in observable_map.items():
+                    if qubit in target:
+                        conflicts = False
+                        if not (
+                            target == measured_qubits
+                            and type(previously_measured) == type(observable)
+                        ):
+                            conflicts = True
+                        elif (
+                            isinstance(observable, Hermitian)
+                            and isinstance(previously_measured, Hermitian)
+                            and not np.allclose(observable.matrix, previously_measured.matrix)
+                        ):
+                            conflicts = True
```

Comment:
Why not just throw errors directly, and include more information about _why_ they occurred?

Reply:
Sure, I can do that. I was just keeping the level of granularity like the BDK